src/leidenalg/CMakeLists.txt

```diff
@@ -0,0 +1,19 @@
+add_library(leidenalg SHARED CPMVertexPartition.cpp
+GraphHelper.cpp
+LinearResolutionParameterVertexPartition.cpp
+ModularityVertexPartition.cpp
+MutableVertexPartition.cpp
+Optimiser.cpp
+RBConfigurationVertexPartition.cpp
+RBERVertexPartition.cpp
+ResolutionParameterVertexPartition.cpp
+SignificanceVertexPartition.cpp
+SurpriseVertexPartition.cpp)
+
+set_target_properties(leidenalg PROPERTIES VERSION ${PROJECT_VERSION})
+set_target_properties(leidenalg PROPERTIES SOVERSION 1)
+if(WIN32 AND NOT CYGWIN)
+  set_target_properties(leidenalg PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS TRUE)
```

Comment:
> > That vcpkg.json is actually a different vcpkg.json than the one in the port. It makes it so that you can just run `vcpkg install` from the root of the directory and it will configure the dependencies itself.
> 
> OK, so this would allow to install to development version so to speak from this repository directly? Users of `vcpkg` would/should still be expected to typically install from the "official" ports, right?

Pretty much. It would let you `git clone` this and run vcpkg install from the root and everything is set up automatically.

Reply:
Windows needs them for the debug build. I think the correct way to do it is to hide it behind a config-time option.

Reply:
Ah-I'm a liar, Windows needs them for shared libraries, period.

Reply:
Oh, I see. Yeah, that sounds like a good idea. Let me know.